storage/inmem/inmem.go

```diff
@@ -92,6 +92,88 @@ func (db *store) NewTransaction(_ context.Context, params ...storage.Transaction
 	return newTransaction(xid, write, context, db), nil
 }
 
+// Truncate implements the storage.Store interface. This method must be called within a transaction.
+func (db *store) Truncate(ctx context.Context, txn storage.Transaction, _ storage.TransactionParams, it storage.Iterator) error {
+	var update *storage.Update
+	var err error
+
+	underlying, err := db.underlying(txn)
+	if err != nil {
+		return err
+	}
+
+	for {
+		update, err = it.Next()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if update.IsPolicy {
+			err = underlying.UpsertPolicy(update.Path, update.Value)
+			if err != nil {
+				return err
+			}
+		} else {
+			p, ok := storage.ParsePathEscaped("/" + update.Path)
+			if !ok {
+				return fmt.Errorf("storage path invalid: %v", update.Path)
+			}
+
+			var obj interface{}
+			err = util.Unmarshal(update.Value, &obj)
+			if err != nil {
+				return err
+			}
+
+			if len(p) > 0 {
+				err = underlying.Write(storage.AddOp, p, obj)
+				if err != nil {
+					return err
+				}
+			} else {
+				// write operation at root path
+
+				if obj == nil {
+					return nil
+				}
```

Comment:
If the update value is null, what happens? I didn't see a test case for this path (might have missed it.) Should this just be caught by the patch error below?

Reply:
If the update path is empty/the write operation is overwriting the root, why do we need anything below? Can't we just make a single Write call into the store?

Reply:
> If the update value is null, what happens? I didn't see a test case for this path (might have missed it.) Should this just be caught by the patch error below?

The test case `TestBundleLazyModeLifecycle` in `bundle/store_test.go` tests this path.  We have some existing tests like `TestBundleLifecycle` in the same file with `Data: nil` and that don't use the truncate call. We don't generate a patch error in those cases. Updated the code a little to remove the `obj == nil` check and directly deserialize into a map. So if the value is `nil`,  resulting map is empty. 


> If the update path is empty/the write operation is overwriting the root, why do we need anything below? Can't we just make a single Write call into the store?

That logic is to handle the scenario where we are activating multiple bundles.